File: src/main/java/kitchenpos/order/tobe/eatinorder/domain/EatInOrderRepository.java
package kitchenpos.order.tobe.eatinorder.domain;

import kitchenpos.order.tobe.eatinorder.domain.ordertable.OrderTable;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface EatInOrderRepository {
    EatInOrder save(EatInOrder eatInOrder);

    Optional<EatInOrder> findById(UUID id);

    List<EatInOrder> findAll();

    boolean existsByOrderTableAndStatusNot(OrderTable orderTable, OrderStatus status);
}

File: src/main/kotlin/kitchenpos/order/tobe/eatinorder/domain/EatInOrderRepository.kt
package kitchenpos.order.tobe.eatinorder.domain

import org.springframework.context.annotation.Primary
import org.springframework.stereotype.Repository

@Repository("tobeEatInOrderRepository")
@Primary
interface EatInOrderRepository {
    fun existsByOrderTableAndStatusNot(orderTable: OrderTable?, status: EatInOrderStatus?): Boolean
}

